internal/command/views/view.go

```diff
@@ -100,6 +105,21 @@ func (v *View) Diagnostics(diags tfdiags.Diagnostics) {
 		return
 	}
 
+	// Filter the deprecation warnings based on the cli arg
+	if v.ModuleDeprecationWarnLvl != tofu.DeprecationWarningLevelAll {
+		var newDiags tfdiags.Diagnostics
+		for _, diag := range diags {
+			if !tofu.DeprecatedOutputDiagAllowed(v.ModuleDeprecationWarnLvl, diag) {
+				continue
+			}
+			if !tofu.DeprecatedVariableDiagAllowed(v.ModuleDeprecationWarnLvl, diag) {
+				continue
+			}
```

Comment:
Do we need to distinguish these two cases with separate functions? It seems like both of these functions are doing essentially the same logic, so I wonder if it would work to have a single function `DeprecationDiagAllowed` which handles both input variable and output value deprecation warnings together. :thinking:

Reply:
Added in 5582eddc3354359f89d8731fce21827a4b109ac7. Thanks for the suggestion.